SUSE/rmt
# Version 2.11 - Mark secrets file as ghost file in rpm spec

## Description

card: https://trello.com/c/JbsuXdpH/2726-rmt-downgrade-210-29-broken

To allow downgrades without file permission errors created due to a missing managed file ('/usr/share/rmt/config/secrets.yml.key) it needs to be tracked as `ghost` file in the rpm spec file.

**How to test this pull request:**

- build the rpm for rmt-server-2.11
```
rmt> make dist
rmt> cp -r obs/package/* <path/to/your/rpm> (or osc co systemsmanagement:SCC:RMT rmt-server)
rpm> build --no-verify --clean SLE_15_SP3 (or what every sp you have docker around)
```

- install and downgrade packages
```
cd /var/tmp/build-root/SLE_15_<YOUR SERVICE PACK!>-x86_64/home/abuild/rpmbuild/RPMS/x86_64

docker run -v (pwd):/rmt --privileged --rm --network=host -h rmt-review -ti connect.15sp3 /bin/bash

> zypper in rmt-server-config-2.11-0.x86_64.rpm rmt-server-2.11-0.x86_64.rpm
> SUSEConnect -r <a regcode of your choice>
> zypper in --oldpackage rmt-server-2.10

#! Check that the installation succeeds without breaking! (aka. permission issues)

```
